### PR TITLE
feat: migrate to jUnit 4

### PR DESCRIPTION
**Pull request by jakubuid**

### CODE REVIEW

File: auth/sdk/src/test/kotlin/com/walletconnect/auth/client/mapper/ClientMapperTest.kt [hunk 1]
```diff
@@ -74,8 +75,8 @@ internal class ClientMapperTest {
             resources = listOf("ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/", "https://example.com/my-web2-claim.json")
         ).toCacaoPayload(iss)
 
-        val sd = SimpleDateFormat(ISO_8601_PATTERN).parse(payload.iat) ?: fail("Cannot parse iat")
-        val iat = Instant.ofEpochMilli(sd.time)
+        val sd = SimpleDateFormat(ISO_8601_PATTERN).parse(payload.iat)
```
Comment (TalhaAli00) [inline auth/sdk/src/test/kotlin/com/walletconnect/auth/client/mapper/ClientMapperTest.kt:78]: Why remove the null check?

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 2]
```diff
@@ -111,25 +111,24 @@ fun DependencyHandlerScope.wsRestJava() {
...
 
-// TODO: Replace JUnit5 with JUnit4 since Android tooling does not support JUnit5 yet
-fun DependencyHandlerScope.jUnit5() {
-    "testImplementation"(platform("org.junit:junit-bom:$jUnit5Version"))
-    "testImplementation"("org.junit.jupiter:junit-jupiter-api")
-    "testRuntimeOnly"("org.junit.jupiter:junit-jupiter-engine")
-    "testImplementation"("org.junit.jupiter:junit-jupiter-params")
+fun DependencyHandlerScope.jUnit4() {
+//    "testImplementation"(platform("org.junit:junit-bom:$jUnit4Version"))
+//    "testImplementation"("org.junit.jupiter:junit-jupiter-api")
```
Comment (TalhaAli00) [inline buildSrc/src/main/kotlin/Dependencies.kt:116]: Anything with Jupiter can removed

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 3]
```diff
@@ -111,25 +111,24 @@ fun DependencyHandlerScope.wsRestJava() {
...
-    "testImplementation"("org.jetbrains.kotlin:kotlin-test-junit5:$kotlinVersion")
+//    "testImplementation"("org.jetbrains.kotlin:kotlin-test-junit5:$kotlinVersion")
 }
 
-fun DependencyHandlerScope.jUnit5Android() {
-    "testImplementation"(platform("org.junit:junit-bom:$jUnit5Version"))
-    "androidTestImplementation"("org.junit.jupiter:junit-jupiter-api")
-    "androidTestRuntimeOnly"("org.junit.jupiter:junit-jupiter-engine")
-}
+//fun DependencyHandlerScope.jUnit5Android() {
```
Comment (TalhaAli00) [inline buildSrc/src/main/kotlin/Dependencies.kt:127]: Let's remove the commented code

Comment (TalhaAli00) [inline buildSrc/src/main/kotlin/Dependencies.kt:127]: Let's remove the commented code.
Also, don't we need JUnit4 for instrumented tests?

**Review by TalhaAli00 — APPROVED** [review]

File: auth/sdk/src/test/kotlin/com/walletconnect/auth/client/mapper/ClientMapperTest.kt (hunk 1 above)
Comment (jakubuid) [reply auth/sdk/src/test/kotlin/com/walletconnect/auth/client/mapper/ClientMapperTest.kt:78]: Unnecessary

File: androidCore/sdk/src/androidTest/kotlin/com/walletconnect/android/test/client/HistoryInstrumentedAndroidTest.kt [hunk 4]
```diff
@@ -13,22 +13,20 @@ import com.walletconnect.android.test.utils.TestClient
...
 import kotlinx.coroutines.runBlocking
-import org.junit.jupiter.api.Assertions
-import org.junit.jupiter.api.Test
-import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.Test
 import timber.log.Timber
 import kotlin.time.Duration.Companion.seconds
 
-@ExtendWith(WCInstrumentedActivityScenario::class)
+//@ExtendWith(WCInstrumentedActivityScenario::class)
```
Comment (kacperoak) [inline androidCore/sdk/src/androidTest/kotlin/com/walletconnect/android/test/client/HistoryInstrumentedAndroidTest.kt:28]: Remove?

**Review by kacperoak — APPROVED** [review]

**Review by Elyniss — APPROVED** [review]
LGTM

**Pull request merged**